### PR TITLE
Add CodeNotary / for version file

### PR DESCRIPTION
**Pull request by pvizeli**
## Proposed change

Currently, we only warn the user. It's like a test before we roll it out.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast supervisor tests`)
- [x] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

File: supervisor/updater.py [hunk 1]
```diff
@@ -180,24 +186,39 @@ async def fetch_data(self):
...
+            raise UpdaterError(
+                f"Can't fetch versions from {url}: {err}", _LOGGER.warning
+            ) from err
 
+        # Validate
+        try:
+            await self.sys_verify_content(checksum=calc_checksum(data))
+        except CodeNotaryUntrusted as err:
+            _LOGGER.critical(
+                "Content-Trust is broaken for the version file fetch! - %s", err
```
Comment (ludeeus) [inline supervisor/updater.py:206]:
```suggestion
                "Content-Trust is broken for the version file fetch! - %s", err
```

File: supervisor/utils/codenotary.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
+"""Small wrapper for haveibeenpwned.com API."""
```
Comment (ludeeus) [inline supervisor/utils/codenotary.py:1]:
```suggestion
"""Small wrapper for CodeNotary."""
```

File: supervisor/utils/codenotary.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            f"Can't parse CodeNotary output: {data!s} - {err!s}", _LOGGER.error
+        ) from err
+
+    if _ATTR_ERROR in data_json:
+        raise CodeNotaryError(data_json[_ATTR_ERROR], _LOGGER.warning)
+
+    if data_json[_ATTR_VERIFICATION][_ATTR_STATUS] == 0:
+        _CACHE.add((checksum, path, org, signer))
+    else:
+        raise CodeNotaryUntrusted(f"{data!s}", _LOGGER.debug)
```
Comment (ludeeus) [inline supervisor/utils/codenotary.py:94]: This is a JSON string, right? any specific parts of that which is useful, I'm guessing the entire object is overkill here?

File: supervisor/utils/pwned.py [hunk 4]
```diff
@@ -38,7 +41,8 @@ async def check_pwned_password(websession: aiohttp.ClientSession, sha1_pw: str)
             return True
 
     except (aiohttp.ClientError, asyncio.TimeoutError) as err:
-        _LOGGER.warning("Can't fetch hibp data: %s", err)
-        raise PwnedConnectivityError() from err
+        raise PwnedConnectivityError(
+            f"Can't fetch hibp data: {err}", _LOGGER.warning
```
Comment (ludeeus) [inline supervisor/utils/pwned.py:45]:
```suggestion
            f"Can't fetch HIBP data: {err}", _LOGGER.warning
```

File: supervisor/utils/codenotary.py (hunk 3 above)
Comment (pvizeli) [reply supervisor/utils/codenotary.py:94]: right, I ditch it

File: supervisor/updater.py [hunk 5]
```diff
@@ -180,24 +186,37 @@ async def fetch_data(self):
...
             _LOGGER.info("Fetching update data from %s", url)
             async with self.sys_websession.get(url, timeout=10) as request:
-                data = await request.json(content_type=None)
+                data = await request.read()
 
         except (aiohttp.ClientError, asyncio.TimeoutError) as err:
             self.sys_supervisor.connectivity = False
-            _LOGGER.warning("Can't fetch versions from %s: %s", url, err)
-            raise UpdaterError() from err
+            raise UpdaterError(
```
Comment (ludeeus) [inline supervisor/updater.py:197]: We should split this.
Timeout error does not have anything for `err`.

**Review by ludeeus — APPROVED** [review]
Code is good :+1: 

It's fine for the dev channel, but there needs to be an option to disable it before it's pushed to beta/stable channels

File: supervisor/updater.py (hunk 5 above)
Comment (pvizeli) [reply supervisor/updater.py:197]: will respect that on next PR which improve that all anyway

**Pull request merged**